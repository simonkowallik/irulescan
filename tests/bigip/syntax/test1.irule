when RULE_INIT priority 500 {
    set n 1;
while {$n < 30}
{
    log local0. [set n [expr $n + 30]]
}
}