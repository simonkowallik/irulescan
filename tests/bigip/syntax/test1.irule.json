[
  {
    "filepath": "test1.irule",
    "warning": [
      {
        "message": "unsafe expression, use braces `{ .. }`",
        "issue_location": "+",
        "context": "expr $n + 30",
        "line": 5
      },
      {
        "message": "unsafe expression, use braces `{ .. }`",
        "issue_location": "30",
        "context": "expr $n + 30",
        "line": 5
      }
    ],
    "dangerous": [
      {
        "message": "dangerous unsafe expression, use braces `{ .. }`",
        "issue_location": "$n",
        "context": "expr $n + 30",
        "line": 5
      }
    ]
  }
]
