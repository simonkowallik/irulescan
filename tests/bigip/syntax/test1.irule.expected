DANGEROUS: Dangerous unquoted expr at `$n` in `expr $n + 30`
WARNING: Unquoted expr at `+` in `expr $n + 30`
WARNING: Unquoted expr at `30` in `expr $n + 30`

